From d31f2a80fca32e2e6d6855841e2ef1539934b361 Mon Sep 17 00:00:00 2001
From: Tomasz Duszynski <tomasz.duszynski@cavium.com>
Date: Mon, 24 Sep 2018 09:04:05 +0200
Subject: [PATCH 0205/1051] net: octeontx2: fix page unmap

SQ and RQ buffers generally differ in size. Fix page unmap by passing
correct size of buffer being unmapped.

Signed-off-by: Tomasz Duszynski <tduszynski@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/otx2_common.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_common.c b/drivers/net/ethernet/marvell/octeontx2/otx2_common.c
index 44a0017a40f8..cc7bd0ca7b43 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_common.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_common.c
@@ -676,17 +676,19 @@ int otx2_config_nix(struct otx2_nic *pfvf)
 
 void otx2_free_aura_ptr(struct otx2_nic *pfvf, int type)
 {
-	int pool_id, pool_start = 0, pool_end = 0;
+	int pool_id, pool_start = 0, pool_end = 0, size = 0;
 	struct otx2_pool *pool;
 	u64 iova, pa;
 
 	if (type == NIX_AQ_CTYPE_SQ) {
 		pool_start = pfvf->hw.rx_queues;
 		pool_end = pfvf->hw.pool_cnt;
+		size = pfvf->hw.sqb_size;
 	}
 	if (type == NIX_AQ_CTYPE_RQ) {
 		pool_start = 0;
 		pool_end = pfvf->hw.rqpool_cnt;
+		size = RCV_FRAG_LEN;
 	}
 
 	/* Free SQB and RQB pointers from the aura pool */
@@ -695,7 +697,7 @@ void otx2_free_aura_ptr(struct otx2_nic *pfvf, int type)
 		iova = otx2_aura_allocptr(pfvf, pool_id);
 		while (iova) {
 			pa = otx2_iova_to_phys(pfvf->iommu_domain, iova);
-			dma_unmap_page_attrs(pfvf->dev, iova, RCV_FRAG_LEN,
+			dma_unmap_page_attrs(pfvf->dev, iova, size,
 					     DMA_FROM_DEVICE,
 					     DMA_ATTR_SKIP_CPU_SYNC);
 			put_page(virt_to_page(phys_to_virt(pa)));
-- 
2.17.1

